Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -486,13 +513,40 @@ public boolean onKeyLongPress(int keyCode, @NonNull KeyEvent event)
   protected void onActivityResult(int requestCode, int resultCode, Intent result)
   {
     super.onActivityResult(requestCode, resultCode, result);
-    if (requestCode == REQUEST_CHANGE_DISC)
+    if (resultCode == RESULT_OK)
     {
-      // If the user picked a file, as opposed to just backing out.
-      if (resultCode == RESULT_OK)
+      if (requestCode == REQUEST_CHANGE_DISC)
       {
+        // If the user picked a file, as opposed to just backing out.
         NativeLibrary.ChangeDisc(result.getData().toString());
       }
+      else if (requestCode == REQUEST_SKYLANDER_FILE)
+      {
+        Pair<Integer, String> slot =
+                SkylanderConfig.loadSkylander(mSkylanderSlots.get(skylanderSlot).getPortalSlot(),
+                        result.getData().toString());
+        clearSkylander(skylanderSlot);
+        mSkylanderSlots.get(skylanderSlot).setPortalSlot(slot.first);
+        mSkylanderSlots.get(skylanderSlot).setLabel(slot.second);
+        mSkylandersBinding.skylandersManager.getAdapter().notifyItemChanged(skylanderSlot);
+        skylanderSlot = -1;
+        skylanderData = new Pair(new Pair<>(-1, -1), "Slot");
```

Comment:
It would be nice to have a named constant for this special (-1, -1) pair. The code as it is right now feels a bit "magic".